### PR TITLE
add connectionToken property from IConnection

### PR DESCRIPTION
**Pull request by wj8400684**

### CODE REVIEW

File: src/SuperSocket.Server/SuperSocketService.cs [hunk 1]
```diff
@@ -314,11 +314,11 @@ protected virtual ValueTask OnSessionClosedAsync(IAppSession session, CloseEvent
             if (closedHandler != null)
                 return closedHandler.Invoke(session, e);
 
-            #if NETSTANDARD2_1
+#if NETSTANDARD2_1
                 return GetCompletedTask();
```
Comment (kerryjiang) [inline src/SuperSocket.Server/SuperSocketService.cs:318]: Indent

File: src/SuperSocket.Server/SuperSocketService.cs [hunk 2]
```diff
@@ -471,20 +467,20 @@ public async Task StartAsync(CancellationToken cancellationToken)
 
         protected virtual ValueTask OnStartedAsync()
         {
-            #if NETSTANDARD2_1
+#if NETSTANDARD2_1
                 return GetCompletedTask();
```
Comment (kerryjiang) [inline src/SuperSocket.Server/SuperSocketService.cs:471]: Indent

File: src/SuperSocket.Server/SuperSocketService.cs [hunk 3]
```diff
@@ -471,20 +467,20 @@ public async Task StartAsync(CancellationToken cancellationToken)
...
+#else
+            return ValueTask.CompletedTask;
+#endif
         }
 
         protected virtual ValueTask OnStopAsync()
         {
-            #if NETSTANDARD2_1
+#if NETSTANDARD2_1
                 return GetCompletedTask();
```
Comment (kerryjiang) [inline src/SuperSocket.Server/SuperSocketService.cs:480]: Indent

**Comment by kerryjiang** [comment]
Also need think about KestrelPipeConnection.

context.ConnectionClosed

https://github.com/wj8400684/SuperSocket/blob/61622a0549c5ffdda8525a92df4a5dc91d8666f5/src/SuperSocket.Kestrel/KestrelPipeConnection.cs#L19

File: src/SuperSocket.Connection/PipeConnectionBase.cs [hunk 4]
```diff
@@ -52,6 +52,7 @@ protected PipeConnectionBase(PipeReader inputReader, PipeWriter outputWriter, Co
             Logger = options.Logger;
             InputReader = inputReader;
             OutputWriter = outputWriter;
+            ConnectionToken = _cts.Token;
         }
 
         protected virtual Task StartTask<TPackageInfo>(IObjectPipe<TPackageInfo> packagePipe)
```
Comment (kerryjiang) [inline src/SuperSocket.Connection/PipeConnectionBase.cs:58]: _cts.Token can be replaced with ConnectionToken in this file. And then we will see how to improve _cts.

Comment (wj8400684) [reply src/SuperSocket.Connection/PipeConnectionBase.cs:58]: image
Is that OK?

Comment (wj8400684) [reply src/SuperSocket.Connection/PipeConnectionBase.cs:58]: image
That should be even better.

**Pull request closed without merging**